pacman-sync: add page (#5268)
tldr-pages/tldr: pages/linux/pacman-sync.md

> # pacman sync

Reviewer: Didn't we decide to use `pacman --sync` as the title?
```suggestion
# pacman --sync
```

Reviewer 2: I think you're thinking of the filename?

Reviewer: No, I'm not, actually.  Don't you think it would be better to have the title match what is actually entered in the terminal?  Currently it looks like `sync` is a normal subcommand of `pacman`.

Reviewer 2: Oh I see what you mean. Yes, I agree - this change would be a good idea.

> Arch Linux package manager utility.
> More information: <https://man.archlinux.org/man/pacman.8>.

- Install a new package:

`sudo pacman --sync {{package_name}}`

> - Synchronize and update all packages (add `--downloadonly` to download only):

Reviewer: How about this?  Coming back to this page after a while, I read it as trying to add the string `--downloadonly` to the string "download only" (???)
```suggestion
- Synchronize and update all packages (add `--downloadonly` to download the packages and not update them):
```

> - Synchronize and update all packages (add `--downloadonly` for download but do not update anything):

Reviewer: Wouldn't it make more sense to _replace_ `--sysupgrade` with `--downloadonly`, rather than add `--downloadonly`?

Author: Without `--sysupgrade`, `pacman` doesn't download the packages. The `--refresh` only updates the local repositories (similar to `apt update`)

Reviewer: Then, how about...
```suggestion
- Synchronize and update all packages (add `--downloadonly` to download only):
```

`pacman --sync --refresh --sysupgrade`

- Search the package database for a regular expression or keyword:

`pacman --sync --search "{{search_pattern}}"`

> - Display information for a package:

Reviewer: ```suggestion
- Display information about a package:
```

`pacman --sync --info {{package_name}}`

> - Install a package as non-explicitly installed:

Author: Maybe is advanced, but orphan packages is a common concept. If you install something as a `dependency` from nothing is an orphan, and there is a `--query` to list all the orphans.

> - Install a package as non-explicitly installed:

Reviewer: The man page says:

>This is useful for makepkg and other build-from-source tools that need to install dependencies before building the package.

But this isn't quite clear to me.  Could you please clarify, and then I could come up with an explanatory description targeted at beginners (from a beginner :smile:)?

> - Install a package as non-explicitly installed:

Author: `makepkg` is used to build packages locally. Arch doens't have `PPA` repositories, you can build one, but is not recommend to the users, security stuff.
The AUR is download the build instruction and build the package locally, then install it.
`makepkg` install build dependencies to build the package.
I use this method when I want to use a package for a short time, and then clean up.

> - Install a package as non-explicitly installed:

Reviewer: If there is another example that would work, I think you should pick it :+1: because that seems more like a power-user concept.

> - Install a package as non-explicitly installed:

Author: Done!

`pacman --sync --asdeps {{package_name}}`

> - Overwrite conflicting files during an update:

Author: There is no system update like `do-release-upgrade`, some arch-based OS's like `Manjaro` try to imitate that behavior.
Sometimes the user needs to "force" replace a file during a package update, the most common scenario is "change of permissions" and the Arch Linux announcements are used for this news.

> - Overwrite conflicting files during an update:

Reviewer: Okay, thank you.  How about changing it to 'package update', because new users might think 'system update' (like I did)
```suggestion
- Overwrite conflicting files during a package update:
```
What do you think?

`pacman --sync --refresh --sysupgrade --overwrite {{path/to/file}}`

> - Synchronize and update all packages except the ignored (can be used more than once):

Reviewer 2: Perhaps this will read better?
```suggestion
- Synchronize and update all packages, but ignore a specific package (can be used more than once):
```

`pacman --sync --refresh --sysupgrade --ignore {{package_name}}`

> - Remove not installed packages and unused repositories from the cache (use `-cc` to clean all packages):

Author: You're right, the docs only said `-cc`.
